### PR TITLE
UnaryExpression struct in Expressions

### PR DESCRIPTION
**Pull request by gzanitti**
This PR is part of issue https://github.com/powdr-labs/powdr/pull/1345.
In particular, it adds the struct UnaryOperation to Expressions to homogenise the structure before including source references.

### CODE REVIEW

File: asm-to-pil/src/vm_to_constrained.rs [hunk 1]
```diff
@@ -538,7 +538,10 @@ impl<T: FieldElement> VMConverter<T> {
                             .filter(|(_, reg)| reg.ty == RegisterTy::Assignment)
                             .map(|(name, _)| rhs_assignment_registers.insert(name.clone()));
                     }
-                    Expression::UnaryOperation(UnaryOperator::Next, e) => {
+                    Expression::UnaryOperation(UnaryOperation {
+                        op: UnaryOperator::Next,
+                        expr: e,
```
Comment (chriseth) [inline asm-to-pil/src/vm_to_constrained.rs:543]:
```suggestion
                        expr,
```

File: asm-to-pil/src/vm_to_constrained.rs [hunk 2]
```diff
@@ -1216,13 +1219,19 @@ fn extract_update(expr: Expression) -> (Option<String>, Expression) {
     };
     // TODO check that there are no other "next" references in the expression
     match *left {
-        Expression::UnaryOperation(UnaryOperator::Next, column) => match *column {
+        Expression::UnaryOperation(UnaryOperation {
+            op: UnaryOperator::Next,
+            expr: column,
```
Comment (chriseth) [inline asm-to-pil/src/vm_to_constrained.rs:1224]:
```suggestion
            expr,
```

File: ast/src/parsed/display.rs [hunk 3]
```diff
@@ -598,7 +598,7 @@ fn format_list<L: IntoIterator<Item = I>, I: Display>(list: L) -> String {
 impl<E: Display> Expression<E> {
     pub fn precedence(&self) -> Option<ExpressionPrecedence> {
         match self {
-            Expression::UnaryOperation(op, _) => Some(op.precedence()),
+            Expression::UnaryOperation(UnaryOperation { op, expr: _ }) => Some(op.precedence()),
             Expression::BinaryOperation(_, op, _) => Some(op.precedence()),
             _ => None,
         }
```
Comment (chriseth) [inline ast/src/parsed/display.rs:604]: I think we should now change the function `format_unary_operation` to just be the Display impl on UnaryOperation.

Comment (gzanitti) [reply ast/src/parsed/display.rs]: Same situation as with BinaryExpression.

**Review by Schaeff — APPROVED** [review]

**Review by chriseth — APPROVED** [review]

**Review by chriseth — APPROVED** [review]

**Pull request merged**